tests/console/yast2_lan_device_settings.pm

```diff
@@ -36,49 +36,57 @@ sub run {
     script_run('ip a');
     script_run('ls -alF /etc/sysconfig/network/');
     save_screenshot;
-
-    my $opened = open_yast2_lan();
-    wait_still_screen;
-    if ($opened eq "Controlled by network manager") {
-        return;
+    unless (check_var("ARCH", "s390x")) {
+        my $opened = open_yast2_lan();
+        wait_still_screen;
+        if ($opened eq "Controlled by network manager") {
+            return;
+        }
+
+        send_key "alt-i";    # open edit device dialog
+        assert_screen 'edit-network-card';
+        send_key "alt-t";    # select static IP address option
+        send_key "tab";
+        type_string $static_ip;
+        send_key 'alt-o';
+        send_key_until_needlematch 'hostname-localhost-removed', 'backspace';
+        type_string $static_hostname;
+        send_key "alt-n";    # next
+        wait_still_screen;
+        if (check_screen "yast_lan_duplicate_ip") {
```

Comment:
Here, I would do:

```perl
assert_screen qw(yast_lan_duplicate_ip static-ip-address-set);
if (match_has_tag 'yast_lan_duplicate_ip') {
   // do some stuff
}
elsif(match_has_tag 'static-ip-address-set') {
  // do another thing
}

```

Because `check_screen` by default has just `0`timeout and it may just not catch the `yast_lan_duplicate_ip` needle.

Reply:
Ahh nice, I was looking in other modules for a way to do this, thanks. I will change it.